pennylane/qnode.py

```diff
@@ -849,6 +858,13 @@ def __call__(self, *args, **kwargs):  # pylint: disable=too-many-branches, too-m
                 # update the gradient function
                 set_shots(self._original_device, override_shots)(self._update_gradient_fn)()
 
+            else:
+                kwargs["shots"] = (
+                    self._original_device._raw_shot_sequence
+                    if self._original_device._shot_vector
+                    else self._original_device.shots
+                )
+

```

Comment:
I wonder if we should move this whole block of code into `construct`, so we only have only place that's dealing with pre-processing the shots.